src/PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.cpp

```diff
@@ -0,0 +1,138 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/InitialData.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace ForceFree::AnalyticData {
+
+RotatingDipole::RotatingDipole(const double vector_potential_amplitude,
+                               const double varpi0, const double delta,
+                               const double angular_velocity,
+                               const double tilt_angle,
+                               const Options::Context& context)
+    : vector_potential_amplitude_(vector_potential_amplitude),
+      varpi0_(varpi0),
+      delta_(delta),
+      angular_velocity_(angular_velocity),
+      tilt_angle_(tilt_angle) {
+  if (varpi0 < 0.0) {
+    PARSE_ERROR(context, "The length constant varpi0 ("
+                             << varpi0_ << ") cannot be negative");
+  }
+  if (delta < 0.0) {
+    PARSE_ERROR(context,
+                "The small number delta (" << delta_ << ") cannot be negative");
+  }
+  if (abs(angular_velocity) >= 1.0) {
+    PARSE_ERROR(context, "The rotation angular velocity ("
+                             << angular_velocity_
+                             << ") must be between -1.0 and 1.0");
+  }
+  if ((tilt_angle < 0.0) or (tilt_angle > M_PI)) {
+    PARSE_ERROR(context, "The rotator tilt angle ("
+                             << tilt_angle_ << ") must be between 0 and Pi");
+  }
+}
+
+RotatingDipole::RotatingDipole(CkMigrateMessage* msg) : InitialData(msg) {}
+
+std::unique_ptr<evolution::initial_data::InitialData>
+RotatingDipole::get_clone() const {
+  return std::make_unique<RotatingDipole>(*this);
+}
+
+void RotatingDipole::pup(PUP::er& p) {
+  InitialData::pup(p);
+  p | vector_potential_amplitude_;
+  p | varpi0_;
+  p | delta_;
+  p | angular_velocity_;
+  p | tilt_angle_;
+  p | background_spacetime_;
+}
+
+PUP::able::PUP_ID RotatingDipole::my_PUP_ID = 0;
+
```

Comment:
We turn on the rotation at t=0 and overwrite `E = B x V` in every time steps, so practically this would not matter.

Reply:
Rotation is turned on at t=0 and `E = B x V` is imposed in every time step, so I believe this doesn't have much practical difference.

Reply:
yes, for this way of evolving the ID, that's totally fine.
This comment (and the next one) are mainly about documentation.

Could you add a comment on this saying that it should be the ideal condition using v, but that this will be imposed during the evolution?